ucan-dns+web.md

```diff
@@ -0,0 +1,209 @@
+
+> Example below illustrates `did:key:zRotation` delegating `did:key:zAli` capabilities derived from `did:dns:w3.storage` through `did:key:zService`. It embeds adove described delegation chain inside fact to provide a verifiable evidence that it can redelegate capbilities on `did:dns:w3.storage` bahalf
+
+```ts
+{
+  iss: "did:key:zRotation",
+  aud: "did:key:zAli",
+  exp: null,
+  att: [
+    {
+      with: "did:key:zAlice",
+      can: "*"
+    }
+  ],
+  prf: [
+    // Proof that did:key:zRotation has authority over
+    // did:dns:w3.storage delegated to it by did:key:zService
+    {
+      iss: "did:key:zService",
+      aud: "did:key:zRotation",
+      exp: null,
+      att: [],
+      // Proof that did:key:zService has been delegated
+      // authority over did:dns:w3.storage as long as
+      // did:key it resolves to is still the one that signed
+      prf: [{
+        iss: "did:dns:w3.storage",
+        aud: "did:key:zService",
+        exp: null
+        att: []
+      }]
+    }
+  ],
+  fct: [
+    // Evidence that "did:dns:w3.storage" has been delegated
+    // full authority over did:key:zAlice by the owner.
+    {
+      iss: "did:key:zAlice",
+      aud: "did:dns:w3.storage",
+      exp: null,
+      att: [
+        {
+          with: "did:key:zAlice",
+          can: "*"
+        }
+      ]
+    },
+  ]
+}
+```
+
+## Delegation of complete authority
+
+[UCAN][] specification does not describe an ability to delegate authority over the resources delegating prinipcal MAY hold in the future. This makes it impossible for `did:dns:w3.storage` key to delegate capability that `did:key:zAlice` will delegate to it in the future.
+
+To overcome this limitation here we propose delegation with `att: []` and `exp: null` to be treated as delegation of complete authority:
+
+```ts
+{
+  iss: "did:dns:w3.storage",
+  aud: "did:key:zService",
+  exp: null
+  att: []
+}
+```
+
+It can also be interpreted as `did:dns:w3.storage` stating to be "also known as"
+`did:key:zService` allowing it to delegate whatever `did:dns:w3.storage` CAN.
+
+## Pipelining dellegations
+
+[UCAN][] specification does not offer a way for two delegation chains to be pipelined into one. In our described scenario we have two delegation chains:
+
+1. `did:key:zAlice -> did:dns:w3.storage`
+2. `did:dns:w3.storage -> did:key:zService -> did:key:zRotation`
+
+From which we would like to construct a delegation of the capability delegated in (1) issued by the the (rightmost) audience in (2).
+
+```
+did:key:zAlice -> did:dns:w3.storage -> did:key:zService -> did:key:zRotation -> did:key:zAli
+```
+
+To accomplish this we propose that:
+
+- To issue delegation from `did:key:zRotation` with a valid proof chain (2) proving that `did:key:zRotation` has been delegated all capabilities from `did:dns:w3.storage`.
+- Embed proof chain (1) in `fct` as a verifable evidence that `did:dns:w3.storage` has been delegated capabilities for `did:key:zAlice`.
```

Comment:
Interesting idea 💡 

This may open up some new avenues for a confused deputy because it relies on any possible UCAN that you learn about in the rest of the network. Arguably this similar to using the ambient authority at the root of a delegation chain, but involves anyone that delegates to the forwarding principal even if they're offline. Aside from phishing, I'll have to think more on specific attacks.

Reply:
Please note that it is implied that:

1. Every delegation is still checked for revocation.
2. `did:dns:w3.storage` resolution still occurs and if it resolves to different key validation will fail.

It is true that you can incorporate any of the delegation you can find on the network, but unless principal it was delegated to redelegated to you it's not possible to utilize it.

Reply:
I think the more that I think about it, the more comfortable I am with it 👍